lib/components/membership-cache.js

```diff
@@ -44,7 +44,12 @@ class MembershipCache {
         if (this._membershipMap[roomId] === undefined) {
             this._membershipMap[roomId] = {};
         }
-        this._registeredUsers.add(userId);
+
+        // Bans and invites do not mean the user exists.
+        if (membership === "join" || membership === "leave") {
```

Comment:
To make the comment true, don't you need this?:
```suggestion
        if (membership === "join" || membership === "leave" || membership === "ban") {
```

Reply:
Nope, we are *only* interested in `join` / `leave`. Not `ban` or `invite`.

Reply:
doh, of course. I can read.